src/storage.js

```diff
@@ -105,3 +75,43 @@ export function ipcStorage(client: Client): AsyncStorage {
       client.request('storage/removeItem', key)
   }
 }
+
+export class StorageSession {
+  _id: string
+  _storage: AsyncStorage
+
+  constructor(id: string, storage: AsyncStorage) {
+    this._id = `${NAMESPACE}-storage-${id}`
+    this._storage = storage
+  }
+
+  async get(key: string): Promise<any> {
+    const data = await this.getData()
+    return key in data ? data[key] : null
+  }
+
+  async getData(): Promise<Object> {
+    const serialized = await this._storage.getItem(this._id)
+    return typeof serialized === 'string' ? JSON.parse(serialized) : {}
```

Comment:
`storage` should probably have a single contract, so always be string? (cfr. `setData` below)?

Reply:
`AsyncStorage.getItem` isn't forced to always return a string, as e.g. localStorage would return null if it doesn't exist. We could wrap it so it always returns a string, but to me this wouldn't seem easy to customize or extend in the future.
I think initializing it with "{}" would be an option. This would make it fail if it gets corrupted in the meantime though (instead of only loosing the login credentials).